### DISCUSSION TITLE
How to accept only few keys as query paramter with any value

### DISCUSSION THREAD

**Discussion by KumarShreyash1** [pillarjs/path-to-regexp#298] [Q&A] [answered]
If someone can please help in a pattern that accepts only specified query parameters with any value. 

Eg: 

Only query a and b is allowed

https://host/path?a=apple&b=banana ---- PASS
https://host/path?a=okay&c=elepahnt&b=cat --- FAIL

**Comment by sohaibaumer** [comment]
You want to allow only the query parameters "a" and "b" in the URL, while ignoring any other parameters. You can achieve it like this in JS:

const allowedParams = ["a", "b"];

function validateURL(url) {
  const urlParams = new URLSearchParams(new URL(url).search);
  for (const param of urlParams.keys()) {
    if (!allowedParams.includes(param)) {
      return false;
    }
  }
  return true;
}

console.log(validateURL("https://host/path?a=apple&b=banana")); // PASS
console.log(validateURL("https://host/path?a=okay&c=elephant&b=cat")); // FAIL

**Reply by KumarShreyash1** [reply, marked as answer]
Hi @sohaibaumer 
Thanks for the answer but I was thinking of a solution using this (path-to-regexp) library itself

**Reply by blakeembrey** [reply]
@KumarShreyash1 this answer is probably the best type of option, this library isn’t intended for unordered data of this kind. There’s a note in the README: “It can not handle arbitrarily ordered data (e.g. query strings, URL fragments, JSON, etc).”
